### ISSUE TITLE
Handle forecasting better (keep forecasted rain and remove it from bucket if it didn't happen. should also move away from the average calc that happens in the bucket)

### ISSUE THREAD

**Issue by akcasoy** [jeroenterheerdt/HAsmartirrigation#381]
### What happened?

So i am testing this integration for a week now, i can say that i am a bit disappointed. I have configured everything exactly as on “Senbei Tristan’s Smartes Heim” german video (with my own wetter station data), although we did not get a lot of rain recently, my bucket keeps increasing every day, and i haven’t seen any positive duration value till now, it keeps 0s.
Most concrete example, the bucket was calculated yesterday night as 6.4mm, and today we had a partly sunny day, more or less 20 degree Celcius, with in total 0.1mm rainrate, my bucket has just been calculated again as 7.0mm. How can it be?
It does not make sense at all. 
I also have a soil moisture sensor, it is working as expected and the soil humidity has been decreased today by 10% for example. But the calculation result of this integration does not work for me, don’t know why.

Thank you for your time in advance!!!

### How to reproduce

Here is my diagnostic file:
[config_entry-smart_irrigation-3d06ed1bcc695c1880eefeb9b105d4d5-3.json](https://github.com/jeroenterheerdt/HAsmartirrigation/files/15447834/config_entry-smart_irrigation-3d06ed1bcc695c1880eefeb9b105d4d5-3.json)

Here is some of my current data from my weather station:
<img width="670" alt="Bildschirmfoto 2024-05-26 um 16 44 32" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/541bf6aa-e7c4-48be-bbec-e3492c7e8ebc">

Daily Rain rate graph for the last 7 days:
<img width="1115" alt="Bildschirmfoto 2024-05-26 um 16 49 57" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/0e66c04b-9866-486f-8336-7e6e9900a680">

Temperatur graph for the last days (mostly sunny and between 20-25 degree C):
<img width="1117" alt="Bildschirmfoto 2024-05-26 um 16 51 02" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/98cceae3-fb8d-4c28-aaa8-476734851e3a">

General tab of the configuration:
<img width="868" alt="Bildschirmfoto 2024-05-26 um 16 53 08" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/7af9fe11-124b-4892-8d79-144f3528143b">

I have 1 zone:
<img width="501" alt="Bildschirmfoto 2024-05-26 um 16 53 31" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/83c923e0-c101-4dd5-8d6d-a322d1fe5dba">

Modules tab of the config:
<img width="588" alt="Bildschirmfoto 2024-05-26 um 16 53 49" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/b8ad8d95-d7ec-4da2-a356-bc991b8a590d">

My Ecowitt sensor group:
<img width="534" alt="Bildschirmfoto 2024-05-26 um 16 54 26" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/637e6c79-c624-4740-96c1-82616e659009">

### Which version are you running?

v2024.4.3

### Diagnostics file

- [X] I have attached a diagnostics file

**Comment by jeroenterheerdt** [comment]
Please open you own issue instead of piling on to another.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by akcasoy** [comment]
I just wanted to inform that the problem continues and can be also seen from the todays values, it may help for the analyse.

Todays rain rate was 0mm:
<img width="561" alt="Bildschirmfoto 2024-05-26 um 23 37 27" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/5ac3421e-1b52-4b56-b766-c08a26f1273a">

Humidity:
<img width="571" alt="Bildschirmfoto 2024-05-26 um 23 38 36" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/9c1f5613-f2f9-49fc-a5af-5a9b85e3952e">

It was mostly a sunny day:
<img width="570" alt="Bildschirmfoto 2024-05-26 um 23 39 15" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/42a8b529-0d51-463f-8dc1-756102ca9e81">

without much wind:
<img width="577" alt="Bildschirmfoto 2024-05-26 um 23 39 50" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/570a1925-9446-42ab-9a7b-2e770bf90561">

Temperature:
<img width="572" alt="Bildschirmfoto 2024-05-26 um 23 40 36" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/aa2f7098-17df-4362-a6f7-5c2e73bd434a">

And my bucket was 7.0mm yesterday, now 6.5mm. So i strangely still do not have to irrigate:
<img width="589" alt="Bildschirmfoto 2024-05-26 um 23 41 09" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/3b188b95-969c-4807-bfd0-ff4fa6200b28">

**Comment by jeroenterheerdt** [comment]
OK, so multiple things here : you are using a 3 day forecast. So of course you can't then look at a days weather and go why did my bucket respond weirdly? Turn off the forecast for now. Also, there is a bug with don't estimate that I am working on to fix. It basically ignores the value from your solrad sensor. Also increase your auto update frequency.

All of those things (two things you can do and one thing that is a bug) should help you get to grips with how this integration works and get better results.

**Comment by jeroenterheerdt** [comment]
So today's calculation is doing perfectly. A bucket decrease of this size is in line with what is expected for this type of weather. How the bucket was so high is likely due to what is wrote above.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by akcasoy** [comment]
So you mean sth. like this?
<img width="542" alt="Bildschirmfoto 2024-05-26 um 23 48 30" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/5bb0871d-56f5-4b6a-8c2b-6d69551422f0">

And:
<img width="369" alt="Bildschirmfoto 2024-05-26 um 23 49 12" src="https://github.com/jeroenterheerdt/HAsmartirrigation/assets/12121451/09a48994-fd16-4da9-8329-4ea52dfa0c47">

**Comment by jeroenterheerdt** [comment]
Yes, like that. Reset your bucket(a) to 0 now and let it run for a couple of days.

**Comment by akcasoy** [comment]
OK thank you.

**Comment by akcasoy** [comment]
I thought a bit about this "3 day forecast" and would like to know why you think that the reason can be that part. 

Does not the logic work this way?: Let's say the bucket was 0 yesterday, and our calculation ended up calculating 3mm today, because with a 80% of chance it will rain a lot tomorrow. So the bucket is 3mm now. 
And tomorrow it did not rain at all.. On the next evening the algorithm should actually remove that part of estimation from yesterdays calculation, and start again from 0, and add to that today's real values from the sensors for example. 

So in that way this 3 day forecast should not have those kind of strange effects on bucket. 

It seem to me that, the bucket was always increasing maybe because the openweather-api were predicting too much rain, and it did not go as expected..

**Comment by jeroenterheerdt** [comment]
That last part is likely what is happening. What we do for forecasting is rather simple: when calculating the et value, include the forecast for x days in advance, and just treat is as one of the daily values. So yeah if owm says it's going to rain, then that is going to bump up the average a lot. Happy to take ideas on what to change / improve, but prefer PRs.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by akcasoy** [comment]
I see that still as a bug. The including weather predictions are one of the main advantages/characteristics of this integration.
But this should work like this:
When the weather prediction calculates, there will be 7mm rain tomorrow. And the evapotranspiration of yesterday was 5mm. The bucket size is calculated as 2mm. 
The next day, when we see that it did not rain at all, 7mm should have been removed from the bucket as first, besides all other calculations for refreshing the bucket value.

**Comment by jeroenterheerdt** [comment]
That's fair, that we can do.

**Comment by jeroenterheerdt** [comment]
We made a lot of changes in this area feel free to open a new issue if needed.
